homeassistant/components/knx/fan.py

```diff
@@ -111,14 +120,21 @@ async def async_turn_on(
         **kwargs: Any,
     ) -> None:
         """Turn on the fan."""
-        if percentage is None:
-            await self.async_set_percentage(DEFAULT_PERCENTAGE)
-        else:
-            await self.async_set_percentage(percentage)
+        speed = percentage
+
+        if self._step_range:
+            if percentage is not None:
+                speed = math.ceil(
+                    percentage_to_ranged_value(self._step_range, percentage)
+                )
+            else:
+                speed = DEFAULT_SPEED_STEP
```

Comment:
to have it perfectly non-breaking we could use 50% as default and run it through `percentage_to_ranged_value()`.
The default constant could be sourced `from xknx.devices.fan import DEFAULT_TURN_ON_SPEED` if you like.

Reply:
The problem is (actually also with my current PR coding) that we once again have defaults in HA, which means xknx does not know for a switch GA fan, if that speed should be set in addition to switching on the fan or whether that was just the default (which should have no impact on switch GA fans).

To really fix that xknx itself would need to contain a copy of `percentage_to_ranged_value()` so that we can also determine what the default speed step for stepped fans should be. Sending that in from HA means we would again incorrectly use that value for switch GA fans.

Or alternatively, we add a new parameter to `self._device.turn_on( )` in xknx so that we can pass both, a speed and a default speed and then xknx can decide what is correct to use.

Reply:
oh dear 😶
I'd not add a default parameter to turn_on - spreading this logic over 2 repositories.

Either add percentage_to_ranged_value() to xknx (would require a new release)
or check for `self._device.switch.initialized` here too to decide if the default should be applied.